packages/x-data-grid/src/components/columnSelection/GridHeaderCheckbox.tsx

```diff
@@ -68,10 +68,15 @@ const GridHeaderCheckbox = React.forwardRef<HTMLButtonElement, GridColumnHeaderP
       // Convert to an object to make O(1) checking if a row exists or not
       // TODO create selector that returns visibleRowIds/paginatedVisibleRowIds as an object
       return rowIds.reduce<Record<GridRowId, true>>((acc, id) => {
+        const isSelectable = apiRef.current.isRowSelectable(id);
+        if (!isSelectable) {
```

Comment:
```suggestion
        if (!apiRef.current.isRowSelectable(id)) {
```
Since you are not using the `const` anywhere else